### PR TITLE
libbeat/instrumentation: Update apm dep to v2.0.0

### PR DESCRIPTION
**Pull request by marclop**
## What does this PR do?

Updates the `go.elastic.co/apm` dependency to `v2.0.0` and the relevant
instrumentation code to use the newer API.

## Why is it important?

Updating old dependencies with neeer ones.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## How to test this PR locally

N/A

## Related issues

https://github.com/elastic/apm-server/pull/7737

## Use cases

N/A

## Screenshots

N/A

## Logs

N/A

### CODE REVIEW

File: libbeat/instrumentation/instrumentation.go [hunk 1]
```diff
@@ -25,17 +25,16 @@ import (
...
+	"go.elastic.co/apm/v2"
+	apmtransport "go.elastic.co/apm/v2/transport"
 
 	"github.com/elastic/beats/v7/libbeat/common"
 	"github.com/elastic/beats/v7/libbeat/common/transport"
 	"github.com/elastic/beats/v7/libbeat/logp"
 )
 
 func init() {
-	// we need to close the default tracer to prevent the beat sending events to localhost:8200
```
Comment (cmacknz) [inline libbeat/instrumentation/instrumentation.go:37]: Was the comment here a lie?

File: libbeat/common/transport/httpcommon/httpcommon.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ import (
 	"net/http"
 	"time"
 
-	"go.elastic.co/apm/module/apmhttp"
+	"go.elastic.co/apm/module/apmhttp/v2"
```
Comment (cmacknz) [inline libbeat/common/transport/httpcommon/httpcommon.go:24]: @kvch we will have to port this to https://github.com/elastic/elastic-agent-libs/blob/df72f6afbb56a8f8547523f63fdd5cb0a8671d69/transport/httpcommon/httpcommon.go#L24

**Comment by cmacknz** [comment]
Some of this code is moving out of the beats repository since it is shared with agent, for example https://github.com/elastic/elastic-agent-libs/blob/main/transport/httpcommon

We've already fixed a lot of the lint warnings there. I should be able to bypass those checks once CI is otherwise green.

**Comment by marclop** [comment]
@cmacknz is the back port to that repository something you folks will do? or should I look into it?

File: libbeat/instrumentation/instrumentation.go (hunk 1 above)
Comment (axw) [reply libbeat/instrumentation/instrumentation.go:37]: We _should_ remove this code.

It used to be that importing `go.elastic.co/apm` would create a default tracer that would actively try to connect to localhost:8200, trying to fetch central config and periodically send metrics. With v2, we now defer creation of this default tracer until `apm.DefaultTracer()` is called.

**Review by axw — CHANGES_REQUESTED** [review]

File: libbeat/instrumentation/instrumentation.go (hunk 1 above)
Comment (marclop) [reply libbeat/instrumentation/instrumentation.go:37]: sounds good, I will remove the DefaultTracer calls.

**Review by axw — APPROVED** [review]

**Comment by marclop** [comment]
I'm not sure why the tests are failing on the CI, but it looks unrelated to my PR, the `main` branch seems to have the same checks failing:

image

**Comment by cmacknz** [comment]
Yes the failures are unrelated to this PR, we're investigating it.

**Comment by marclop** [comment]
@cmacknz I'm not authorized to merge the PR and from my end looks good to merge, but I leave it up to you folks to decide when you want to merge it.

**Comment by cmacknz** [comment]
@Mergifyio update

**Comment by cmacknz** [comment]
Packaging failed for reasons that have nothing to do with the changes in this PR it seems:

```
[2022-04-05T18:09:55.170Z] ERROR: generateBuildReport: Error generating build report
[2022-04-05T18:09:55.175Z] groovy.lang.MissingMethodException: No signature of method: net.sf.json.JSONNull.contains() is applicable for argument types: (java.lang.String) values: [githubPrCheckApproved]
[2022-04-05T18:09:55.175Z] Possible solutions: toString(), toString(), toString(), notify(), toString(int)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.CpsDefaultGroovyMethods.findAll(CpsDefaultGroovyMethods.java:1068)
[2022-04-05T18:09:55.175Z] Caused: groovy.text.TemplateExecutionException: Template execution error at line 82:
[2022-04-05T18:09:55.175Z]         81:                                       !it?.displayName?.contains('Archive JUnit') &&
[2022-04-05T18:09:55.175Z]     --> 82:                                       !it?.displayDescription?.contains('githubPrCheckApproved') &&
[2022-04-05T18:09:55.175Z]         83:                                       !it?.displayDescription?.contains('approval-list/elastic') &&
[2022-04-05T18:09:55.175Z] 
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.sandbox.DefaultInvoker.methodCall(DefaultInvoker.java:20)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.impl.ContinuationGroup.methodCall(ContinuationGroup.java:86)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.dispatchOrArg(FunctionCallBlock.java:113)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.impl.FunctionCallBlock$ContinuationImpl.fixName(FunctionCallBlock.java:78)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.impl.ContinuationPtr$ContinuationImpl.receive(ContinuationPtr.java:72)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.impl.ConstantBlock.eval(ConstantBlock.java:21)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.Next.step(Next.java:83)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:174)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.Continuable$1.call(Continuable.java:163)
[2022-04-05T18:09:55.175Z] 	at com.cloudbees.groovy.cps.Continuable.run0(Continuable.java:163)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.access$001(SandboxContinuable.java:18)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.SandboxContinuable.run0(SandboxContinuable.java:51)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.CpsThread.runNextChunk(CpsThread.java:185)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.run(CpsThreadGroup.java:403)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup.access$400(CpsThreadGroup.java:97)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:315)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.CpsThreadGroup$2.call(CpsThreadGroup.java:279)
[2022-04-05T18:09:55.175Z] 	at org.jenkinsci.plugins.workflow.cps.CpsVmExecutorService$2.call(CpsVmExecutorService.java:67)
[2022-04-05T18:09:55.175Z] 	at hudson.remoting.SingleLaneExecutorService$1.run(SingleLaneExecutorService.java:139)
[2022-04-05T18:09:55.175Z] 	at jenkins.util.ContextResettingExecutorService$1.run(ContextResettingExecutorService.java:28)
[2022-04-05T18:09:55.175Z] 	at jenkins.security.ImpersonatingExecutorService$1.run(ImpersonatingExecutorService.java:68)
```

**Review by cmacknz — APPROVED** [review]

**Pull request merged**